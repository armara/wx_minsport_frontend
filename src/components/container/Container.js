import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import ToolTipIcon from 'components/toolTip/icon/ToolTipIcon'

import './Container.scss'

const Container = ({ title, className, children, dataTip }) => (
  <div className={classnames(['base-container', className])}>
    {title && (
      <h5 className="base-container_title">
        {title}
        {dataTip && <ToolTipIcon dataTip={dataTip} />}
      </h5>
    )}
    {children}
  </div>
)

Container.defaultProps = {
  title: '',
  className: '',
  dataTip: '',
}

Container.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
  dataTip: PropTypes.string,
}

export default Container
