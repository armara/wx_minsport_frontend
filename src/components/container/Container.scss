@import 'styles/mixins';

.base-container {
  display: flex;
  flex-flow: column;
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 20px;
  border: 1px solid $smoke-white;
  border-radius: 6px;

  .base-container_title {
    @include container-header-text();
    margin-left: -20px;
    width: calc(100% + 40px);
    padding-left: 10px;

    & > .tooltip-icon {
      margin-top: 2px;
    }
  }
}
