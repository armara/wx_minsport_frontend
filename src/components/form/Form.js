import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { Form as FinalForm } from 'react-final-form'
import arrayMutators from 'final-form-arrays'

import 'components/form/Form.scss'

const Form = ({
  validate,
  onSubmit,
  decorators,
  initialValues,
  subscription,
  className,
  children,
  childrenProps,
  ...rest
}) => (
  <FinalForm
    onSubmit={onSubmit}
    validate={validate}
    decorators={decorators}
    initialValues={initialValues}
    mutators={{
      ...arrayMutators,
    }}
  >
    {({ handleSubmit, form }) => {
      return (
        <form
          onSubmit={handleSubmit}
          className={classnames([className, 'base-form'])}
        >
          {typeof children === 'function'
            ? children({ form, initialValues, ...childrenProps, ...rest })
            : children}
        </form>
      )
    }}
  </FinalForm>
)

Form.defaultProps = {
  className: '',
  initialValues: {},
  childrenProps: {},
  subscription: {},
  decorators: [],
  validate: () => {},
}

Form.propTypes = {
  decorators: PropTypes.arrayOf(PropTypes.func),
  subscription: PropTypes.shape({}),
  className: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  validate: PropTypes.func,
  childrenProps: PropTypes.shape({}),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.func,
  ]).isRequired,
  initialValues: PropTypes.shape({}),
}

export default Form
