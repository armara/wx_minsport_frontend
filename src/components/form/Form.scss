@import "styles/mixins";

.base-form {
  display: flex;
  flex-flow: column;

  h4 {
    @include form-label-text();
  }
}

.form-some-blocks {
  display: flex;

  & > div {
    flex-grow: 1;
  }

  & > div + div {
    margin-left: 10px;
  }
}
