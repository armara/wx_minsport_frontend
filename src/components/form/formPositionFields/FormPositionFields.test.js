import React from 'react'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Position from 'components/form/formPositionFields/FormPositionFields'
import Form from 'components/form/Form'
import Container from 'components/container/Container'
import initialState from 'tests/initialStoreMock'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

configure({ adapter: new Adapter() })
const mockStore = configureStore()

describe('<Position />', () => {
  let store
  let component

  beforeEach(() => {
    // creates the store with any initial state or middleware needed
    store = mockStore(initialState)
    localStorage.setItem('user', '{"test":"ok"}')

    const coachForEdit = {
      positions: [null],
    }
    component = (
      <Provider store={store}>
        <Form onSubmit={() => {}} initialValues={coachForEdit}>
          <Container title="Должности">
            <Position
              allowToSelectPosition
              positions={[
                { id: '0', title: 'Pos1', type: 'MANAGER' },
                { id: '1', title: 'Pos2', type: 'SPORT' },
              ]}
            />
          </Container>
        </Form>
      </Provider>
    )
  })

  it('Should render correctly', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should change discipline on sport select', () => {
    const wrapper = mount(component)
    const content = wrapper.find('.dropdown-content')

    const sportItems = content.at(0).find('button')
    const hockeyButton = sportItems.at(0)
    const soccerButton = sportItems.at(1)
    const discButton = content.at(1).find('button')

    // given
    expect(discButton.text()).toBe('Любая дисциплина')
    expect(hockeyButton.text()).toBe('Хоккей')
    expect(soccerButton.text()).toBe('Футбол')

    const checkDisciplineItems = expectedItems => {
      wrapper.update()
      const discItems = wrapper
        .find('.dropdown-content')
        .at(1)
        .find('button')
      discItems.forEach((item, index) =>
        expect(item.text()).toBe(expectedItems[index])
      )
    }

    // when click on soccer
    soccerButton.simulate('click')
    checkDisciplineItems(['Любая дисциплина', 'футбол'])

    // when click on hockey
    hockeyButton.simulate('click')
    checkDisciplineItems(['Любая дисциплина', 'хоккей'])
  })

  it('Should render positions from props', () => {
    const wrapper = mount(component)
    const content = wrapper.find('.dropdown-content')
    const positionButtons = content.at(2).find('button')
    const pos1 = positionButtons.at(0)
    const pos2 = positionButtons.at(1)

    expect(pos1.text()).toBe('Pos1')
    expect(pos2.text()).toBe('Pos2')
  })
})
