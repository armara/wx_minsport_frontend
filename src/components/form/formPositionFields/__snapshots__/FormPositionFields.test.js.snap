// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Position /> Should render correctly 1`] = `
<form
  className="base-form"
  onSubmit={[Function]}
>
  <div
    className="base-container"
  >
    <h5
      className="base-container_title"
    >
      Должности
    </h5>
    <div
      className="form-positions-field"
    >
      <div
        className="form-field form-field_required"
      >
        <label
          htmlFor="positions[0].sport"
        >
          Вид спорта
        </label>
        <div
          className="form-field_with-error"
        >
          <div
            className="form-filter-dropdown-with-value"
          >
            <div
              className="dropdown"
              id="position-sport-0"
            >
              <div
                className="dropdown-trigger"
              >
                <p
                  className="control has-icons-right has-icons-left input-trigger"
                >
                  <input
                    className="is-small base-input input"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Введите вид спорта"
                    type="text"
                    value=""
                  />
                  <span
                    className="icon is-small is-left"
                  >
                    <i
                      className="ic-place"
                    />
                  </span>
                  <span
                    className="icon is-small is-right"
                  >
                    <i
                      className="ic-keyboard-arrow-down"
                    />
                  </span>
                </p>
              </div>
              <div
                className="dropdown-menu"
                role="menu"
              >
                <ul
                  className="dropdown-content menu-list"
                >
                  <button
                    className="dropdown-item button base-button"
                    disabled={false}
                    id={null}
                    onClick={[Function]}
                    title=""
                    type="button"
                  >
                    
                    Хоккей
                  </button>
                  <button
                    className="dropdown-item button base-button"
                    disabled={false}
                    id={null}
                    onClick={[Function]}
                    title=""
                    type="button"
                  >
                    
                    Футбол
                  </button>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-field"
      >
        <label
          htmlFor="positions[0].discipline"
        >
          Дисциплина
        </label>
        <div
          className=""
        >
          <div
            className="form-filter-dropdown-with-value"
          >
            <div
              className="dropdown"
              id="position-discipline-0"
            >
              <div
                className="dropdown-trigger"
              >
                <p
                  className="control has-icons-right has-icons-left input-trigger"
                >
                  <input
                    className="is-small base-input input"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Введите название дисциплины"
                    type="text"
                    value=""
                  />
                  <span
                    className="icon is-small is-left"
                  >
                    <i
                      className="ic-place"
                    />
                  </span>
                  <span
                    className="icon is-small is-right"
                  >
                    <i
                      className="ic-keyboard-arrow-down"
                    />
                  </span>
                </p>
              </div>
              <div
                className="dropdown-menu"
                role="menu"
              >
                <ul
                  className="dropdown-content menu-list"
                >
                  <button
                    className="dropdown-item button base-button"
                    disabled={false}
                    id={null}
                    onClick={[Function]}
                    title=""
                    type="button"
                  >
                    
                    Любая дисциплина
                  </button>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-field form-field_required"
      >
        <label
          htmlFor="positions[0].position"
        >
          Должность в спорте
        </label>
        <div
          className="form-field_with-error"
        >
          <div
            className="form-filter-dropdown-with-value"
          >
            <div
              className="dropdown"
              id="position-position-0"
            >
              <div
                className="dropdown-trigger"
              >
                <p
                  className="control has-icons-right has-icons-left input-trigger"
                >
                  <input
                    className="is-small base-input input"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Введите название должности в спорте"
                    type="text"
                    value=""
                  />
                  <span
                    className="icon is-small is-left"
                  >
                    <i
                      className="ic-place"
                    />
                  </span>
                  <span
                    className="icon is-small is-right"
                  >
                    <i
                      className="ic-keyboard-arrow-down"
                    />
                  </span>
                </p>
              </div>
              <div
                className="dropdown-menu"
                role="menu"
              >
                <ul
                  className="dropdown-content menu-list"
                >
                  <button
                    className="dropdown-item button base-button"
                    disabled={false}
                    id={null}
                    onClick={[Function]}
                    title=""
                    type="button"
                  >
                    
                    Pos1
                  </button>
                  <button
                    className="dropdown-item button base-button"
                    disabled={false}
                    id={null}
                    onClick={[Function]}
                    title=""
                    type="button"
                  >
                    
                    Pos2
                  </button>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr />
      <button
        className="inline-button button base-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        title=""
        type="button"
      >
        
        Добавить должность
      </button>
    </div>
  </div>
</form>
`;
